integrations/google_genai/src/haystack_integrations/components/embedders/google_genai/text_embedder.py

```diff
@@ -0,0 +1,138 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Dict, List, Optional, Union
+
+from google import genai
+from google.genai import types
+from haystack import component, default_from_dict, default_to_dict, logging
+from haystack.utils import Secret, deserialize_secrets_inplace
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class GoogleGenAITextEmbedder:
+    """
+    Embeds strings using Google AI models.
+
+    You can use it to embed user query and send it to an embedding Retriever.
+
+    ### Usage example
+
+    ```python
+    from haystack_integrations.components.embedders.google_genai import GoogleGenAITextEmbedder
+
+    text_to_embed = "I love pizza!"
+
+    text_embedder = GoogleGenAITextEmbedder()
+
+    print(text_embedder.run(text_to_embed))
+
+    # {'embedding': [0.017020374536514282, -0.023255806416273117, ...],
+    # 'meta': {'model': 'text-embedding-004-v2',
+    #          'usage': {'prompt_tokens': 4, 'total_tokens': 4}}}
+    ```
+    """
+
+    def __init__(  # pylint: disable=too-many-positional-arguments
+        self,
+        *,
+        api_key: Secret = Secret.from_env_var("GOOGLE_API_KEY"),
+        model: str = "text-embedding-004",
+        prefix: str = "",
+        suffix: str = "",
+        config: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        Creates an GoogleGenAITextEmbedder component.
+
+        :param api_key:
+            The Google API key.
+            You can set it with the environment variable `GOOGLE_API_KEY`, or pass it via this parameter
+            during initialization.
+        :param model:
+            The name of the model to use for calculating embeddings.
+            The default model is `text-embedding-004`.
+        :param prefix:
+            A string to add at the beginning of each text to embed.
+        :param suffix:
+            A string to add at the end of each text to embed.
+        :param config:
+            A dictionary of keyword arguments to configure embedding content configuration `types.EmbedContentConfig`.
+            For more information, see the [Google AI Task types](https://ai.google.dev/gemini-api/docs/embeddings#task-types).
```

Comment:
Could we mention here that if none is specified we set it to `{"task_type": "SEMANTIC_SIMILARITY"}` by default?